[Woo POS][Product labels] Update simple products to have stock properties

iamgabrielma opened the pull request:
<!-- Remember about a good descriptive title. -->

Closes: WOOMOB-488

## Description
This PR updates the `POSProduct` and `POSSimpleProduct` models to contain the inventory-related properties we'll need to display stock quantities and stock status, that is `manageStock`, `stockQuantity`, and `stockStatusKey`.

Variations will be handled separately to keep the PRs small.

## Testing information
There are no functional changes, and the properties are not used yet anywhere, just smoke testing POS to confirm nothing has changed should be enough. CI should pass.

I'll be AFK next week, so absolutely no rush to review this. Feel free to merge it directly if all is good, otherwise I can pick it up when I'm back 👍 

---

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

iamgabrielma committed:
update POSProduct with stock properties

iamgabrielma committed:
Update POSSimpleProduct with stock properties

dangermattic commented:
<!--
  0 Errors
  1 Warning: View files have been modified,...
  0 Messages
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

iamgabrielma committed:
make test target compile

staskus commented on Networking/Networking/Model/POSProduct.swift:
```diff
@@ -35,6 +35,10 @@ public struct POSProduct: Codable, Equatable, GeneratedCopiable, GeneratedFakeab
 
+    public let manageStock: Bool
+    public let stockQuantity: Decimal?
+    public let stockStatusKey: String
```
```suggestion
    public let stockStatusKey: String
```

Based [on documentation](https://woocommerce.github.io/woocommerce-rest-api-docs/#list-all-products):

> `stock_status`  Options: instock, outofstock and onbackorder.

I think we should just encode it it directly into an enum, it will be easier to work with it.

iamgabrielma replied:
Yeah, that's something I did on a first iteration until I saw we might expect different values that the ones shared in the API docs based on different plugins, or at least we do so in the existing stock implementation in the app side. 

We can see this in `ProductStockStatus` where they handle it with two additional cases:

```
case insufficientStock // Product Bundles only
case custom(String) // in case there are extensions modifying product stock statuses
```

At the moment I'm not sure how this will affect stock in POS, since we do not support bundles, but the custom status might do some funny business for simple products/variations. I've logged this [here](https://linear.app/a8c/issue/WOOMOB-517/woo-posproduct-labels-handle-undocumented-stock-statuses) to follow-up separately

staskus reviewed (approved):
Looks good. Only suggestion is to turn stock_status directly into enum for easier usability.

iamgabrielma committed:
Merge branch 'trunk' into task/woomob-488-update-pos-product-for-inventory

iamgabrielma committed:
resolve make compile

After merging trunk into the branch this doesnt seem to compile, there’s an extra comma in analytics, possibly from a merge conflict somewhere else.

iamgabrielma committed:
make test compile
